pkg/blockissuer/blockissuer.go

```diff
@@ -95,31 +180,19 @@ func (i *BlockIssuer) CreateBlock(ctx context.Context, p iotago.Payload, parents
 }
 
 // CreateBlockWithReferences creates a new block with the given payload and parent references.
-func (i *BlockIssuer) CreateBlockWithReferences(ctx context.Context, p iotago.Payload, references model.ParentReferences, strongParentsCountOpt ...int) (*model.Block, error) {
-	var err error
-	references, err = i.getReferences(ctx, p, references, strongParentsCountOpt...)
-	if err != nil {
-		return nil, errors.WithMessagef(err, "tipselection failed, error: %s", err.Error())
-	}
-
-	slotCommitment := i.protocol.MainEngineInstance().Storage.Settings().LatestCommitment()
-	lastFinalizedSlot := i.protocol.MainEngineInstance().Storage.Settings().LatestFinalizedSlot()
-
-	block, err := builder.NewBlockBuilder().
-		StrongParents(references[model.StrongParentType]).
-		WeakParents(references[model.WeakParentType]).
-		ShallowLikeParents(references[model.ShallowLikeParentType]).
-		SlotCommitment(slotCommitment.Commitment()).
-		LatestFinalizedSlot(lastFinalizedSlot).
-		Payload(p).
-		Sign(i.Account.ID(), i.Account.PrivateKey()).
-		ProofOfWork(ctx, float64(i.protocol.MainEngineInstance().Storage.Settings().ProtocolParameters().MinPoWScore)).
-		Build()
+func (i *BlockIssuer) CreateBlockWithReferences(ctx context.Context, p iotago.Payload, references model.ParentReferences, strongParentsCountOpt ...int) (modelBlock *model.Block, err error) {
```

Comment:
Why do we still need this function and `CreateBlock`? Can't we simply model this with `CreateBlockWithOptions` and rename it to `CreateBlock` ? Would be cleaner in my opinion

Reply:
done